### DISCUSSION TITLE
how to store data from rest api directly to duckdb  wasm without only on the client site

### DISCUSSION THREAD

**Discussion by kzmKZuzmo** [observablehq/framework#1644] [General]
Hello is there a way to fetch data from rest api and store tham directly to duckdbwasm  without touching the server **!!!ONLY ON CLIENT SITE!!!**

**Comment by Fil** [comment]
Yes, see https://observablehq.com/framework/sql

This example is client-side only:

```
---
sql:
  quakes: https://earthquake.usgs.gov/earthquakes/feed/v1.0/summary/all_day.csv
---
```

Or, you can use any of DuckDB's commands, for example `read_csv()`.

**Reply by kzmKZuzmo** [reply]
@Fil but I need create some Authorization (custom logic) how should I handle that ?

**Reply by Fil** [reply]
That's a question for duckdb. See https://duckdb.org/docs/api/wasm/data_ingestion for an overview of all the mechanisms to ingest data. Note that if you need particular credentials to connect to the API, you will need to pass credentials to the user's browser. If you must keep the credentials secret, you have to use data loaders.
